pkg/cmd/roachtest/tests/cluster_to_cluster.go

```diff
@@ -67,11 +70,68 @@ type c2cSetup struct {
 	metrics      c2cMetrics
 }
 
+var c2cPromMetrics = map[string]clusterstats.ClusterStat{
+	"LogicalMegabytes": {
+		LabelName: "node",
+		Query:     "replication_logical_bytes / 1e6"},
+	"PhysicalMegabytes": {
+		LabelName: "node",
+		Query:     "replication_sst_bytes / 1e6"},
+	"PhysicalReplicatedMegabytes": {
+		LabelName: "node",
+		Query:     "capacity_used / 1e6"},
+}
+
+func sumOverLabel(stats map[string]map[string]clusterstats.StatPoint, label string) float64 {
+	var mean float64
+	for _, stat := range stats[label] {
+		mean += stat.Value
+	}
+	return mean
+}
+
+func (cc *c2cSetup) startStatsCollection(
+	ctx context.Context, t test.Test, c cluster.Cluster,
+) func(time.Time) map[string]float64 {
+
+	if c.IsLocal() {
+		// Grafana does not run locally.
+		return func(snapTime time.Time) map[string]float64 {
+			return map[string]float64{}
+		}
+	}
+	// TODO(msbutler): pass a proper cluster replication dashboard and figure out why we need to
+	// pass a grafana dashboard for this to work
+	cfg := (&prometheus.Config{}).
+		WithPrometheusNode(cc.workloadNode.InstallNodes()[0]).
+		WithCluster(cc.dst.nodes.InstallNodes()).
+		WithNodeExporter(cc.dst.nodes.InstallNodes()).
+		WithGrafanaDashboard("https://go.crdb.dev/p/changefeed-roachtest-grafana-dashboard")
+
+	require.NoError(t, c.StartGrafana(ctx, t.L(), cfg))
+	t.L().Printf("Prom has started")
+
+	client, err := clusterstats.SetupCollectorPromClient(ctx, c, t.L(), cfg)
+	require.NoError(t, err, "error creating prometheus client for stats collector")
+	collector := clusterstats.NewStatsCollector(ctx, client)
+
+	return func(snapTime time.Time) map[string]float64 {
+		metricSnap := make(map[string]float64)
+		for name, stat := range c2cPromMetrics {
+			point, err := collector.CollectPoint(ctx, t.L(), snapTime, stat.Query)
+			if err != nil {
+				t.L().Errorf("Could not query prom %s", err.Error())
+			}
+			metricSnap[name] = sumOverLabel(point, stat.LabelName)
```

Comment:
Is it possible to put this summation into the Query like you can in the grafana graph?

Reply:
Sadly, not with this `collector.CollectPoint()` api. I'll leave a todo and file an issue.